pyocd/target/builtin/target_AMA3B1KK.py

```diff
@@ -0,0 +1,211 @@
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000
+    ],
+
+    # Relative function addresses
+    'pc_init': 0x10000021,
+    'pc_unInit': 0x10000033,
+    'pc_program_page': 0x10000083,
+    'pc_erase_sector': 0x10000065,
+    'pc_eraseAll': 0x10000037,
+
+    'static_base' : 0x10000000 + 0x00000020 + 0x000000f4,
+    'begin_stack' : 0x10000400,
+    'begin_data' : 0x10000000 + 0x1000,
+    'page_size' : 0x2000,
+    'analyzer_supported' : False,
+    'analyzer_address' : 0x00000000,
+    'page_buffers' : [0x10001000, 0x10003000],   # Enable double buffering
+    'min_program_length' : 0x2000,
+
+    # Flash information
+    'flash_start': 0xc000,
+    'flash_size': 0xF4000,
+    'sector_sizes': (
+        (0x0, 0x2000),
+    )
+}
+
+class Apollo3(CortexM):
+    MCUCTRL_SCRATCH0 = 0x400401B0
+    MCUCTRL_BOOTLDR = 0x400401A0
+    JDEC_PID = 0xF0000FE0
+
+    def __init__(self, session, ap, memoryMap, core_num, acquire_timeout):
+        self._acquire_timeout = acquire_timeout
+        super(Apollo3, self).__init__(session, ap, memoryMap, core_num)
+
+    def set_reset_catch(self, reset_type=None):
+        """! @brief Prepare to halt core on reset."""
+        #LOG.debug("set reset catch, core %d", self.core_number)
+
+        self._reset_catch_delegate_result = self.call_delegate('set_reset_catch', core=self, reset_type=reset_type)
+        
+        # Default behaviour if the delegate didn't handle it.
+        if not self._reset_catch_delegate_result:
+            # Halt the target.
+            self.halt()
+
+        jdecpid = self.read_memory(self.JDEC_PID)
+        #print("JDEC PID " + str(jdecpid))
+
+        if((jdecpid & 0xf0) == 0xc0):
+            #print("Ambiq Apollo3")
+            bootldr = self.read_memory(self.MCUCTRL_BOOTLDR)
+            if ((bootldr & 0x0C000000) == 0x04000000):
```

Comment:
Done. This was honestly a bit of a struggle since I was just using magic numbers i got from ambiq, but i was able to piece it a bit from the datasheet. Let me know if you think they still need improvement, I could reach out to ambiq for more information.